grpc/creds/creds.go

```diff
@@ -10,20 +10,23 @@ import (
 	"google.golang.org/grpc/credentials"
 )
 
-// transportCredentials is a grpc/credentials.TransportCredentials which supports
+// clientTransportCredentials is a grpc/credentials.TransportCredentials which supports
 // connecting to, and verifying multiple DNS names
-type transportCredentials struct {
+type clientTransportCredentials struct {
 	roots   *x509.CertPool
 	clients []tls.Certificate
 }
 
-// New returns a new initialized grpc/credentials.TransportCredentials
-func New(rootCAs *x509.CertPool, clientCerts []tls.Certificate) credentials.TransportCredentials {
-	return &transportCredentials{rootCAs, clientCerts}
+// NewClientTransport returns a new initialized grpc/credentials.TransportCredentials for client usage
+func NewClientTransport(rootCAs *x509.CertPool, clientCerts []tls.Certificate) credentials.TransportCredentials {
```

Comment:
This and the counter part for the server-side have slightly confusing naming, could we go with `New...Credentials` instead?